packages/viewer/src/components/viewer-markup-circle/viewer-markup-circle.tsx

```diff
@@ -0,0 +1,338 @@
+import {
+  Component,
+  Host,
+  h,
+  Element,
+  Prop,
+  Watch,
+  EventEmitter,
+  Event,
+  State,
+  Method,
+} from '@stencil/core';
+import { Point, Rectangle } from '@vertexvis/geometry';
+import { DeviceSize, getDeviceSize } from '../../lib/device';
+import { getMouseClientPosition } from '../../lib/dom';
+import {
+  BoundingBox2dAnchorPosition,
+  translateRectToScreen,
+  translatePointToRelative,
+} from '../viewer-markup/utils';
+import {
+  BoundingBox2d,
+  SvgShadow,
+} from '../viewer-markup/viewer-markup-components';
+import { parseBounds, transformCircle } from './utils';
+import { getMarkupBoundingClientRect } from '../viewer-markup/dom';
+
+/**
+ * The supported markup modes.
+ *
+ * @see {@link ViewerMarkupCircleMode.mode} - For more details about modes.
+ */
+export type ViewerMarkupCircleMode = 'edit' | 'replace' | '';
+
+@Component({
+  tag: 'vertex-viewer-markup-circle',
+  styleUrl: 'viewer-markup-circle.css',
+  shadow: true,
+})
+export class ViewerMarkupCircle {
+  /**
+   * The bounds of the circle. Can either be an instance of a `Rectangle` or
+   * a JSON string representation in the format of `[x, y, width, height]` or
+   * `{"x": 0, "y": 0, "width": 10, "height": 10}`.
+   *
+   * Bounds are expected to have relative coordinates, with `[x, y]` from `[-0.5, 0.5]`
+   * and `[width, height]` from `[0, 1]`, e.g. `[0, 0, 0.25, 0.25]`corresponds to a circle
+   * with a diameter of one fourth the viewport's smallest size in the center of the viewport.
+   */
+  @Prop({ mutable: true, attribute: null })
+  public bounds?: Rectangle.Rectangle;
+
+  /**
+   * The bounds of the circle. Can either be an instance of a `Rectangle` or
+   * a JSON string representation in the format of `[x, y, width, height]` or
+   * `{"x": 0, "y": 0, "width": 0.1, "height": 0.1}`.
+   *
+   * Bounds are expected to have relative coordinates, with `[x, y]` from `[-0.5, 0.5]`
+   * and `[width, height]` from `[0, 1]`, e.g. `[0, 0, 0.25, 0.25]`corresponds to a circle
+   * with a diameter of one fourth the viewport's smallest size in the center of the viewport.
+   */
+  @Prop({ attribute: 'bounds' })
+  public boundsJson?: string;
+
+  /**
+   * A mode that specifies how the markup component should behave. When
+   * unset, the component will not respond to interactions with the handles.
+   * When `edit`, the markup anchors are interactive and the user is able
+   * to reposition them. When `replace`, anytime the user clicks on the canvas,
+   * a new markup will be performed.
+   */
+  @Prop({ reflect: true })
+  public mode: ViewerMarkupCircleMode = '';
+
+  /**
+   * The viewer to connect to markups.
+   *
+   * This property will automatically be set when a child of a
+   * `<vertex-viewer-markup>` or `<vertex-viewer>` element.
+   */
+  @Prop()
+  public viewer?: HTMLVertexViewerElement;
+
+  /**
+   * An event that is dispatched anytime the user begins editing the
+   * markup.
+   */
+  @Event({ bubbles: true })
```

Comment:
IIRC the `bubbles` value defaults to true, so this is redundant.

Reply:
Not sure if this is the case with Stencil. Would be good to 2x check.

Reply:
Sounds good. I was wrong :)